apps/dgst.c

```diff
@@ -414,13 +414,54 @@ int dgst_main(int argc, char **argv)
     return ret;
 }
 
+/*
+ * The newline_escape_filename function performs newline escaping for any 
+ * filename that contains a newline.  This function also takes a pointer 
+ * to backslash. The backslash pointer is a flag to indicating whether a newline
+ * is present in the filename.  If a newline is present, the backslash flag is 
+ * set and the output format will contain a backslash at the beginning of the
+ * digest output. This output format is to replicate the output format found
+ * in the '*sum' checksum programs. This aims to preserve backward 
+ * compatibility.
+ */
+static const char *newline_escape_filename(const char *file, int * backslash)
+{
+    size_t i = 0, e = 0, length = strlen(file), newline_count = 0, mem_len = 0;
+    char *file_cpy = NULL;
+    while(i < length) {
+        const char c = file[i];
+        if (c == '\n') {
+            newline_count++;
+        }
+        i++;
+    }
+    mem_len = length + newline_count;
```

Comment:
`+ 1` for the \0 terminator.

This one isn't optional.

Reply:
i.e. `mem_len = length + newline_count + 1;`

Reply:
Great catch thank you.  Pushing this up in my new commit.